### PR TITLE
v9 Update: First MVC Series - 10 tutorials + app samples

### LINKED ISSUE #32708: .NET 9 Update: First MVC Tutorials Series and improvements
### Description

- [ ] Update the entire series and code samples for .NET 9.
- [ ] Consider how we might address the problem of having two commands in a row to copy/paste, but the 2nd only runs if you notice to hit the return key again such as for Add-Migration and Update-Database.

Rick suggested considering an innocuous command at the end such as:
`Echo Update complete`.
`Get-Date`
`$null = $null`
`Write-Host "commands complete"`

Luke considered splitting the command instruction and instead of grouping the information on what the commands do, separating them between the two commands for each. 

There is also the possibility of joining commands with && so an additional return is not required but that might cause some confusion. See 5-23-24 9:02 thread discussion. 

Check in with what Luke decided to do for the same issue for Blazor. The same problem exists for the Razor Pages tutorial series as well.

### Page URL

https://learn.microsoft.com/en-us/aspnet/core/tutorials/first-mvc-app/start-mvc?view=aspnetcore-8.0&tabs=visual-studio

### Content source URL

https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/tutorials/first-mvc-app/start-mvc.md

### Document ID

46a7c2e0-4871-491f-02c2-54f7df2237b1

### Article author

@wadepickett

---
[Associated WorkItem - 261116](https://dev.azure.com/msft-skilling/Content/_workitems/edit/261116)

### PR DESCRIPTION
**Pull request by wadepickett**
Fixes #32708

- Updated all 10 tutorials in the series.
- Updated app samples.
- Updated screenshots.
- Removed all Visual Studio for Mac specific instructions/tabs.
- Fixed some existing minor issues.
- Updated some includes.

- I split up `Add-Migration InitialCreate` and `Update-Database` so we are no longer asking to run them together since many people were not realizing they still needed to hit return one more time after fixating on the expected warning.
[Part 3, add a view to an ASP.NET Core MVC app](https://review.learn.microsoft.com/en-us/aspnet/core/tutorials/first-mvc-app/adding-model?view=aspnetcore-9.0&branch=pr-en-us-33192&tabs=visual-studio#initial-migration)

I did a side by side visual comparison plus diff comparison for each tutorial and tested through the instructions and code for each step to completion.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/tutorials/first-mvc-app/adding-controller.md](https://github.com/dotnet/AspNetCore.Docs/blob/af95b7116b5bfc75fe1530440b85b0528cfe9eef/aspnetcore/tutorials/first-mvc-app/adding-controller.md) | [Part 2, add a controller to an ASP.NET Core MVC app](https://review.learn.microsoft.com/en-us/aspnet/core/tutorials/first-mvc-app/adding-controller?branch=pr-en-us-33192) |
| [aspnetcore/tutorials/first-mvc-app/adding-model.md](https://github.com/dotnet/AspNetCore.Docs/blob/af95b7116b5bfc75fe1530440b85b0528cfe9eef/aspnetcore/tutorials/first-mvc-app/adding-model.md) | [[Visual Studio](#tab/visual-studio)](https://review.learn.microsoft.com/en-us/aspnet/core/tutorials/first-mvc-app/adding-model?branch=pr-en-us-33192) |
| [aspnetcore/tutorials/first-mvc-app/adding-view.md](https://github.com/dotnet/AspNetCore.Docs/blob/af95b7116b5bfc75fe1530440b85b0528cfe9eef/aspnetcore/tutorials/first-mvc-app/adding-view.md) | [[Visual Studio](#tab/visual-studio)](https://review.learn.microsoft.com/en-us/aspnet/core/tutorials/first-mvc-app/adding-view?branch=pr-en-us-33192) |
| [aspnetcore/tutorials/first-mvc-app/controller-methods-views.md](https://github.com/dotnet/AspNetCore.Docs/blob/af95b7116b5bfc75fe1530440b85b0528cfe9eef/aspnetcore/tutorials/first-mvc-app/controller-methods-views.md) | [Part 6, controller methods and views in ASP.NET Core](https://review.learn.microsoft.com/en-us/aspnet/core/tutorials/first-mvc-app/controller-methods-views?branch=pr-en-us-33192) |
| [aspnetcore/tutorials/first-mvc-app/details.md](https://github.com/dotnet/AspNetCore.Docs/blob/af95b7116b5bfc75fe1530440b85b0528cfe9eef/aspnetcore/tutorials/first-mvc-app/details.md) | [Part 10, examine the Details and Delete methods of an ASP.NET Core app](https://review.learn.microsoft.com/en-us/aspnet/core/tutorials/first-mvc-app/details?branch=pr-en-us-33192) |
| [aspnetcore/tutorials/first-mvc-app/new-field.md](https://github.com/dotnet/AspNetCore.Docs/blob/af95b7116b5bfc75fe1530440b85b0528cfe9eef/aspnetcore/tutorials/first-mvc-app/new-field.md) | [[Visual Studio / Visual Studio for Mac](#tab/visual-studio+visual-studio-mac)](https://review.learn.microsoft.com/en-us/aspnet/core/tutorials/first-mvc-app/new-field?branch=pr-en-us-33192) |
| [aspnetcore/tutorials/first-mvc-app/search.md](https://github.com/dotnet/AspNetCore.Docs/blob/af95b7116b5bfc75fe1530440b85b0528cfe9eef/aspnetcore/tutorials/first-mvc-app/search.md) | [aspnetcore/tutorials/first-mvc-app/search](https://review.learn.microsoft.com/en-us/aspnet/core/tutorials/first-mvc-app/search?branch=pr-en-us-33192) |
| [aspnetcore/tutorials/first-mvc-app/start-mvc.md](https://github.com/dotnet/AspNetCore.Docs/blob/af95b7116b5bfc75fe1530440b85b0528cfe9eef/aspnetcore/tutorials/first-mvc-app/start-mvc.md) | [Get started with ASP.NET Core MVC](https://review.learn.microsoft.com/en-us/aspnet/core/tutorials/first-mvc-app/start-mvc?branch=pr-en-us-33192) |
| [aspnetcore/tutorials/first-mvc-app/validation.md](https://github.com/dotnet/AspNetCore.Docs/blob/af95b7116b5bfc75fe1530440b85b0528cfe9eef/aspnetcore/tutorials/first-mvc-app/validation.md) | [Part 9, add validation to an ASP.NET Core MVC app](https://review.learn.microsoft.com/en-us/aspnet/core/tutorials/first-mvc-app/validation?branch=pr-en-us-33192) |
| [aspnetcore/tutorials/first-mvc-app/working-with-sql.md](https://github.com/dotnet/AspNetCore.Docs/blob/af95b7116b5bfc75fe1530440b85b0528cfe9eef/aspnetcore/tutorials/first-mvc-app/working-with-sql.md) | [Part 5, work with a database in an ASP.NET Core MVC app](https://review.learn.microsoft.com/en-us/aspnet/core/tutorials/first-mvc-app/working-with-sql?branch=pr-en-us-33192) |

### CODE REVIEW

File: aspnetcore/tutorials/first-mvc-app/start-mvc.md [hunk 1]
```diff
@@ -30,19 +29,21 @@ At the end of the series, you'll have an app that manages and displays movie dat
 
 [View or download sample code](https://github.com/dotnet/AspNetCore.Docs/tree/main/aspnetcore/tutorials/first-mvc-app/start-mvc/sample) ([how to download](xref:index#how-to-download-a-sample)).
 
+At the end of the series, you'll have an app that manages, validates and displays movie data. 
+
+![Completed application showing the search UI](~/tutorials/first-mvc-app/search/_static/9/search2_90.png)
+
+![Completed application showing client side valiation features](~/tutorials/first-mvc-app/validation/_static/9/val90.png)
```
Comment (tdykstra) [inline aspnetcore/tutorials/first-mvc-app/start-mvc.md:36]: Consider omitting this; at this point in the tutorial it is a big block of clutter to have to scroll through.

File: aspnetcore/tutorials/first-mvc-app/start-mvc.md [hunk 2]
```diff
@@ -30,19 +29,21 @@ At the end of the series, you'll have an app that manages and displays movie dat
 
 [View or download sample code](https://github.com/dotnet/AspNetCore.Docs/tree/main/aspnetcore/tutorials/first-mvc-app/start-mvc/sample) ([how to download](xref:index#how-to-download-a-sample)).
 
+At the end of the series, you'll have an app that manages, validates and displays movie data.
```
Comment (tdykstra) [inline aspnetcore/tutorials/first-mvc-app/start-mvc.md:32]:
```suggestion
At the end of the series, you'll have an app that manages, validates, and displays movie data. 
```

File: aspnetcore/tutorials/first-mvc-app/start-mvc.md [hunk 3]
```diff
@@ -30,19 +29,21 @@ At the end of the series, you'll have an app that manages and displays movie dat
...
+[!INCLUDE[](~/includes/net-prereqs-vs-9.0.md)]
 
 # [Visual Studio Code](#tab/visual-studio-code)
 
-[!INCLUDE[](~/includes/net-prereqs-vsc-8.0.md)]
-
-# [Visual Studio for Mac](#tab/visual-studio-mac)
-
-[!INCLUDE[](~/includes/net-prereqs-mac-8.0.md)]
+[!INCLUDE[](~/includes/net-prereqs-vsc-9.0.md)]
```
Comment (tdykstra) [inline aspnetcore/tutorials/first-mvc-app/start-mvc.md:46]: Have you verified that the instructions work with C# Dev Kit?

File: aspnetcore/tutorials/first-mvc-app/controller-methods-views.md
Comment (tdykstra) [inline aspnetcore/tutorials/first-mvc-app/controller-methods-views.md:1]: Pre-existing issue: the other pages say what the name of the page that the Next and Previous buttons go to, after "Next:" and "Previous:" This one just has "Next" and "Previous" on the buttons. For consistency you could take the page name off the ones that have it or add it to the ones that don't have it. Or ignore the issue, it isn't critical.

**Review by tdykstra — APPROVED** [review]
This looks great, I had just a few comments.

File: aspnetcore/tutorials/first-mvc-app/start-mvc.md (hunk 3 above)
Comment (wadepickett) [reply aspnetcore/tutorials/first-mvc-app/start-mvc.md:46]: Good point. I did not.  It was blocking the prompt for creating assets and could not be used as of late last year per my notes.  I will test the latest with C# Dev Kit reinstalled.

Comment (wadepickett) [reply aspnetcore/tutorials/first-mvc-app/start-mvc.md:40]: Tested now:  Yes it does work.  Assets are just added without the dialog that used to stop the works and then keep the add assets UI from showing.  Looks like it was also added back into the prereqs already.  So nothing more to change.  Thanks for bringing it up.

**Pull request merged**